File: Exemplos/App_eXperience_Java/Elgin_AppExperience_Java-font/app/src/main/java/com/elgin/elginexperience/BalancaPage.java
package com.elgin.elginexperience;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Spinner;
import android.widget.TextView;

import com.elgin.elginexperience.Services.BalancaService;

import java.util.HashMap;
import java.util.Map;

public class BalancaPage extends AppCompatActivity {
    public BalancaService balanca;

    TextView textReturnValueBalanca;

    RadioGroup radioGroupModels;
    RadioButton radioButtonDP30CK;

    Spinner spinnerProtocols;

    Button buttonConfigurarBalanca;
    Button buttonLerPeso;

    String typeModel = "DP30CK";
    String typeProtocol = "PROTOCOL 0";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_balanca_page);

        balanca = new BalancaService(this);

        textReturnValueBalanca = findViewById(R.id.textReturnValueBalanca);
        radioButtonDP30CK = findViewById(R.id.radioButtonDP30CK);
        spinnerProtocols = findViewById(R.id.spinnerProtocols);
        buttonConfigurarBalanca = findViewById(R.id.buttonConfigurarBalanca);
        buttonLerPeso = findViewById(R.id.buttonLerPeso);


        //CONFIGS MODEL BALANÇA
        radioButtonDP30CK.setChecked(true);
        radioGroupModels = findViewById(R.id.radioGroupModels);
        radioGroupModels.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @SuppressLint("NonConstantResourceId")
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                if (checkedId == R.id.radioButtonDP30CK) {
                    typeModel = "DP30CK";
                } else if (checkedId == R.id.radioButtonSA110) {
                    typeModel = "SA110";
                } else if (checkedId == R.id.radioButtonDPSC) {
                    typeModel = "DPSC";
                }
            }
        });

        //CONFIGS PROTOCOLS
        spinnerProtocols.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onNothingSelected(AdapterView<?> parent) { }

            @Override
            public void onItemSelected(AdapterView adapter, View v, int i, long lng) {
                typeProtocol = adapter.getItemAtPosition(i).toString();
            }
        });

        buttonConfigurarBalanca.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sendConfigBalanca();
            }
        });

        buttonLerPeso.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { sendLerPesoBalanca(); }
        });
    }

    public void sendConfigBalanca(){
        Map<String, Object> mapValues = new HashMap<>();
        mapValues.put("model", typeModel);
        mapValues.put("protocol", typeProtocol);
        
        balanca.configBalanca(mapValues);
    }

    public void sendLerPesoBalanca(){
        String retorno = balanca.lerPesoBalanca();
        textReturnValueBalanca.setText(retorno);
    }
}

File: Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/BalancaPage.kt
package com.example.e1_elgin_kotlin

import android.os.Bundle
import android.view.View
import android.widget.*
import android.widget.AdapterView.OnItemSelectedListener
import androidx.appcompat.app.AppCompatActivity
import java.util.*


class BalancaPage : AppCompatActivity() {
    var balanca: Balanca? = null
    var textReturnValueBalanca: TextView? = null
    lateinit var radioGroupModels: RadioGroup
    lateinit var radioButtonDP30CK: RadioButton
    lateinit var spinnerProtocols: Spinner
    lateinit var buttonConfigurarBalanca: Button
    lateinit var buttonLerPeso: Button
    var typeModel = "DP30CK"
    var typeProtocol = "PROTOCOL 0"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_balanca_page)
        balanca = Balanca(this)
        textReturnValueBalanca = findViewById(R.id.textReturnValueBalanca)
        radioButtonDP30CK = findViewById(R.id.radioButtonDP30CK)
        spinnerProtocols = findViewById(R.id.spinnerProtocols)
        buttonConfigurarBalanca = findViewById(R.id.buttonConfigurarBalanca)
        buttonLerPeso = findViewById(R.id.buttonLerPeso)


        //CONFIGS MODEL BALANÇA
        radioButtonDP30CK.setChecked(true)
        radioGroupModels = findViewById(R.id.radioGroupModels)
        radioGroupModels.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, checkedId ->
            when (checkedId) {
                R.id.radioButtonDP30CK -> typeModel = "DP30CK"
                R.id.radioButtonSA110 -> typeModel = "SA110"
                R.id.radioButtonDPSC -> typeModel = "DPSC"
            }
        })

        //CONFIGS PROTOCOLS
        spinnerProtocols.setOnItemSelectedListener(object : OnItemSelectedListener {
            override fun onNothingSelected(parent: AdapterView<*>?) {}
            override fun onItemSelected(adapter: AdapterView<*>, v: View, i: Int, lng: Long) {
                typeProtocol = adapter.getItemAtPosition(i).toString()
            }
        })
        buttonConfigurarBalanca.setOnClickListener(View.OnClickListener { sendConfigBalanca() })
        buttonLerPeso.setOnClickListener(View.OnClickListener { sendLerPesoBalanca() })
    }

    fun sendConfigBalanca() {
        val mapValues: MutableMap<String?, Any?> = HashMap()
        mapValues["model"] = typeModel
        mapValues["protocol"] = typeProtocol
        balanca!!.configBalanca(mapValues)
    }

    fun sendLerPesoBalanca() {
        val retorno = balanca!!.lerPesoBalanca()
        textReturnValueBalanca!!.text = retorno
    }
}